pkg/controller/machine_safety_test.go

```diff
@@ -290,18 +291,48 @@ var _ = Describe("#machine_safety", func() {
 
 			machineSets, err := c.machineSetLister.List(labels.Everything())
 			Expect(err).To(BeNil())
+
+			// Validate MachineSets with Setup Data
 			Expect(len(machineSets)).To(Equal(1))
 
+			for annotation := range data.setup.machineSetAnnotations {
+				Expect(data.setup.machineSetAnnotations[annotation]).To(Equal(machineSets[0].Annotations[annotation]))
+			}
+
+			for label := range data.setup.machineSetLabels {
+				Expect(data.setup.machineSetLabels[label]).To(Equal(machineSets[0].Labels[label]))
+			}
+
+			if len(machineSets[0].Status.Conditions) > 0 {
+				Expect(data.setup.machineSetConditions[0]).To(Equal(machineSets[0].Status.Conditions[0]))
+			}
+
+			// Validate MachineDeployment with the Setup Data
+			machineDeployments, err := c.machineDeploymentLister.List(labels.Everything())
+			Expect(err).To(BeNil())
+
+			for annotation := range data.setup.machineDeploymentAnnotations {
+				Expect(data.setup.machineDeploymentAnnotations[annotation]).To(Equal(machineDeployments[0].Annotations[annotation]))
+			}
+
+			for label := range data.setup.machineDeploymentLabels {
+				Expect(data.setup.machineDeploymentLabels[label]).To(Equal(machineDeployments[0].Labels[label]))
+			}
+
+			if len(machineDeployments[0].Status.Conditions) > 0 {
+				Expect(data.setup.machineDeploymentConditions[0]).To(Equal(machineDeployments[0].Status.Conditions[0]))
+			}
+
```

Comment:
```suggestion
			for k, v := range data.setup.machineSetAnnotations {
				Expect(machineSets[0].Annotations).To(HaveKeyWithValue(k, v))
			}

			for k, v := range data.setup.machineSetLabels {
				Expect(machineSets[0].Labels).To(HaveKeyWithValue(k, v))
			}

			if len(machineSets[0].Status.Conditions) > 0 {
				Expect(machineSets[0].Status.Conditions).To(ContainElements(data.setup.machineSetConditions...))
			}

			// Validate MachineDeployment with the Setup Data
			machineDeployments, err := c.machineDeploymentLister.List(labels.Everything())
			Expect(err).To(BeNil())

			for k, v := range data.setup.machineDeploymentAnnotations {
				Expect(machineDeployments[0].Annotations).To(HaveKeyWithValue(k, v))
			}

			for k, v := range data.setup.machineDeploymentLabels {
				Expect(machineDeployments[0].Labels).To(HaveKeyWithValue(k, v))
			}

			if len(machineDeployments[0].Status.Conditions) > 0 {
				Expect(machineDeployments[0].Status.Conditions).To(ContainElements(data.setup.machineDeploymentConditions...))
			}

```

Reply:
Adopted this suggestion with a minor change of typecasting the matcher to type `[]v1.MachineDeploymentConditions` instead of variable length interface.